packages/core/contracts/cross-chain-oracle/chain-adapters/Admin_ChildMessenger.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../interfaces/ChildMessengerInterface.sol";
+import "../interfaces/ChildMessengerConsumerInterface.sol";
+import "../../common/implementation/Lockable.sol";
+
+/**
+ * @notice A version of the child messenger that allows an admin to relay messages on its behalf.
+ * @dev No parent messenger is needed for this case, as the admin could be trusted to manually send DVM requests on
+ * mainnet. This is intended to be used as a "beta" deployment compatible with any EVM-compatible chains before
+ * impleenting a full bridge adapter. Put simply, it is meant as a stop-gap.
+ */
+contract Admin_ChildMessenger is Ownable, Lockable, ChildMessengerInterface {
+    // The only child network contract that can send messages over the bridge via the messenger is the oracle spoke.
+    address public oracleSpoke;
+
+    event SetOracleSpoke(address newOracleSpoke);
+    event MessageSentToParent(bytes data);
+    event MessageReceivedFromParent(bytes data, address indexed targetSpoke, address indexed caller);
+
+    /**
+     * @notice Changes the stored address of the Oracle spoke, deployed on L2.
+     * @dev The caller of this function must be the admin.
+     * @param newOracleSpoke address of the new oracle spoke, deployed on L2.
+     */
+    function setOracleSpoke(address newOracleSpoke) public onlyOwner nonReentrant() {
+        oracleSpoke = newOracleSpoke;
+        emit SetOracleSpoke(newOracleSpoke);
+    }
+
+    /**
+     * @notice Logs a message to be manually relayed to L1.
+     * @dev The caller must be the OracleSpoke on L2. No other contract is permissioned to call this function.
+     * @param data data message sent to the L1 messenger. Should be an encoded function call or packed data.
+     */
+    function sendMessageToParent(bytes memory data) public override nonReentrant() {
+        require(msg.sender == oracleSpoke, "Only callable by oracleSpoke");
+
+        // The data formmat is kept the same to match the format in other ChildMessenger classes.
+        bytes memory dataSentToParent = abi.encodeWithSignature("processMessageFromCrossChainChild(bytes)", data);
```

Comment:
this part is a bit redundant as the use of this contract is to decode whatever was sent from the spoke and then manually relay information to L1. I guess it is good to keep the data's encoding the exact same as the other spokes with the encoded function call on `processMessageFromCrossChainChild`.

Reply:
I agree this is weird. I decided to change it to just log out the data. Thoughts?

Reply:
ye, I think that's fine. the only real complexity comes in when we are publishing prices. we need to encode things correctly but that's honestly not too difficult given the simplicity of the `processMessageFromParent` on the `OracleSpoke` and `OracleHub.